test/ToolsTestFileIO.cpp

```diff
@@ -954,6 +935,92 @@ TEST_F(TestFileIO, DISABLED_System_Performance_FileIO_DirectoryReader__vs_Boost_
    timeCheck = timeToFind.ElapsedMs();
    std::cout  << "Boost filesystem found           " << boostFileCounter << " items in: " 
        << timeCheck << " millisec" << std::endl;
+}
 
 
+TEST_F(TestFileIO, CreateOneFileAndReadItIn) {
+   // Create subdirectory: /tmp/TempDirectoryTestOfFileIO/TestDir/
+   std::string newTestDirName = "TestDir";
+   std::string newDirectoryPath = mTestDirectory + std::string("/") + newTestDirName;
+   std::string createdDirectoryPath = CreateSubDirectory(newTestDirName, mTestDirectory);
+   EXPECT_TRUE(FileIO::DoesDirectoryExist(newDirectoryPath));
+   EXPECT_EQ(newDirectoryPath, createdDirectoryPath);
+
+   // Add 1 file
+   std::vector<std::string> filenames = {"test1"};
+   for (const auto& filename : filenames) {
+      std::string createdFilePath = CreateFile(createdDirectoryPath, filename);
+      EXPECT_TRUE(FileIO::DoesFileExist(createdFilePath));
+   }
+
+   // Read directory contents
+   auto dirContentsResult = FileIO::GetDirectoryContents(createdDirectoryPath);
+   EXPECT_TRUE(dirContentsResult.HasSuccess());
+   auto vecOfDirContents = dirContentsResult.result; 
+   EXPECT_EQ(vecOfDirContents.size(), filenames.size());
+   for (const auto& filename : vecOfDirContents) {
+      EXPECT_TRUE(std::find(vecOfDirContents.begin(), vecOfDirContents.end(), filename) != vecOfDirContents.end());
+   }
 }
+
+TEST_F(TestFileIO, CreateMultipleFilesAndReadThemIn) {
+   // Create subdirectory: /tmp/TempDirectoryTestOfFileIO/TestDir/
+   std::string newTestDirName = "TestDir";
+   std::string newDirectoryPath = mTestDirectory + std::string("/") + newTestDirName;
+   std::string createdDirectoryPath = CreateSubDirectory(newTestDirName, mTestDirectory);
+   EXPECT_TRUE(FileIO::DoesDirectoryExist(newDirectoryPath));
+   EXPECT_EQ(newDirectoryPath, createdDirectoryPath);
+
+   // Add multiple files
+   std::vector<std::string> filenames = {"test1", "test2", "test3", "test4"};
```

Comment:
Can you please merge the functionality to a function that takes a input and runs all the logic?  It seems that the functions are almost identical. 

Then the tests would be minimalistic and very easy to read
```
TEST_F(TestFileIO, CreateMultipleFilesAndReadThemIn) {
   auto filenames = std::vector<std::string>{...};
    CreateDirectoryContent(path, filenames);
   auto contentResult = FileIO::GetDirectoryContents(path);
   ASSERT_TRUE(contentREsult.hasSuccess());
   EXPECT_TRUE(VerifyDirectoryContent(filenames, contentResult.result);
}
```